### ISSUE TITLE
Bind component's children into `children` prop instead of using `componentChildrenContainer`

### ISSUE THREAD

**Issue by mohasarc** [codelab-app/platform#3272]
## Description

Currently the way children are made available to the component using `componentChildrenContainer` by specifying an element within the component to hold any child passed to the component when it's being used.

This behaviour is limiting as it doesn't allow for the children to be used in multiple places within the component. We can utilize the prop binding feature to generalize this case by attaching the children passed to a component to a prop called `children` in the component's interface.

The final behaviour would look something like this (in terms of React)

```js
const Component = ({children}) => {
   // here we can use children however we like
}
```

```js
<Page>
     <Component>
          <SomeElement/>
    </Component>
</Page>
```

So basically, we only need to make `SomeElement` available for `Component` throught the `children` prop. Then within the component, we can bind that prop to any element using #3268

**Comment by webberwang** [comment]
For Component/Atom api we will create a non-deletable children key.

Thinking we add as data not hardcoded

**Comment by mohasarc** [comment]
how can we make it non-deletable if we're not hardcoding it?

**Comment by webberwang** [comment]
> how can we make it non-deletable if we're not hardcoding it?

Good point, I suppose we could create a `PropsWithChildren` api, then we add it to all the atom api's programmatically as Union.

But in reality it's easier to just hardcode it

**Comment by assimhabeek** [comment]
via Hubstaff
User: assim

Project: platform - https://app.hubstaff.com/projects/2961404
Date Range: 04/04/24 - 04/09/24
Work session total: 4:02:39
Billable: Yes

Grand total: 4:02:39
